handlers/java_handler/src/main/java/mercator/MavenUtils.java

```diff
@@ -79,8 +82,47 @@ public static Document getParsedExpandedPom(File pomFile) {
             return null;
         }
 
-        Document parsedPom = readFileAsDocument(resolvedPom);
-        return parsedPom;
+        try {
+            File resolvedPomFixed = removeDuplicateXMLTag(resolvedPom);
+            Document parsedPom = readFileAsDocument(resolvedPomFixed);
+            return parsedPom;
+        } catch (java.io.IOException ex) {
+            return null;
+        }
+
+    }
+
+    /**
+     * Remove duplicate XML start tag from given XML file.
+     * @param resolvedPom
+     * @return File handle having only one XML start tag
+     * @throws IOException
+     */
+    public static File removeDuplicateXMLTag(File resolvedPom) throws IOException {
+        File resolvedPomFixed;
+        resolvedPomFixed = File.createTempFile("resolvedpom-fixed", ".xml");
+        resolvedPomFixed.deleteOnExit();
+
+        Scanner sc = new Scanner(new FileInputStream(resolvedPom));
+        String xmlTag = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>";
+        int xmlTagCount = 0;
+
+        PrintWriter pw = new PrintWriter(resolvedPomFixed);
+        while(sc.hasNextLine()) {
```

Comment:
Any reason not to use `replace` instead of iterating through the whole file?

Reply:
Can you elaborate on how big files can be? I don't really think this optimization is really worth it, but I am curious how big these files can be.

Reply:
I can't give size on how big the files can be because that is project specific and also depends on how how git is the pom hierarchy for that project.

NOTE: The current approach is not an optimization. It is the natural way to filter data in a file line by line. Using `replace` will be inefficient than this.

Reply:
Let's not continue here, because going line by line is not a bad approach. 

What I have tried to suggest is that there is no need to discuss efficiency here, because files will be up to 10Mb which is basically nothing. I find using `replace` far more readable that iterating through the whole file.